dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -536,4 +558,24 @@ private String gitHubAppSetup(String installationId) {
         }
         return installationAccessToken;
     }
+
+    /**
+     * Parse git references (ex. refs/heads/feature/foobar) and returns a pair of the branch type and name
+     * @param gitReference Git Reference of the form refs/tags/name or refs/heads/name
+     * @return Pair of reference information
+     */
+    private Pair<String, String> parseGitHubReference(String gitReference) {
+        // Match the github reference (ex. refs/heads/feature/foobar or refs/tags/1.0)
+        Pattern pattern = Pattern.compile("^refs/(tags|heads)/([a-zA-Z0-9]+([./_-]?[a-zA-Z0-9]+)*)$");
+        Matcher matcher = pattern.matcher(gitReference);
+
+        if (!matcher.find()) {
+            String msg = "Reference " + gitReference + " is not of the valid form";
+            LOG.error(msg);
+            throw new CustomWebApplicationException(msg, LAMBDA_FAILURE);
+        }
+        String gitBranchType = matcher.group(1);
```

Comment:
This is going to be  the string`tags` or `heads` isn't it? Shouldn't you convert this to a Version.ReferenceType? 

But see my above comments -- you aren't actually using this. And if you do need to use it, then maybe a Pair<ReferenceType, String> is OK instead of creating a new class.

Reply:
I agree, but decided to just change the function to return the git reference name.